azure-quantum/azure/quantum/target/rigetti.py

```diff
@@ -0,0 +1,100 @@
+"""Defines targets and helper functions for the Rigetti provider"""
+
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+##
+
+__all__ = [
+    "Readout",
+    "Result",
+    "Rigetti",
+    "RigettiTarget",
+]
+
+import json
+from enum import Enum
+from typing import Union, Dict, List, TypeVar, Generic, cast
+
+from .target import Target
+from ..job import Job
+from ..workspace import Workspace
+
+
+class RigettiTarget(str, Enum):
+    """The known targets for the Rigetti provider
+
+    See https://qcs.rigetti.com/qpus for details on a QPU target.
+    """
+
+    QVM = "rigetti.qvm"
+    """A simulator target for Quil. See https://github.com/quil-lang/qvm for more info."""
+
+    ASPEN_11 = "rigetti.aspen-11"
```

Comment:
Could we document the number of qubits?

Reply:
We could, but there is going to be manual Markdown documentation outside of this with more detail. I'd expect that folks will use that to pick their QPU and just use this enum as a convenience.

The reason a raw string is accepted is because the delay between a new target being added and a new SDK release could be >1 month IIUC.